compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -3371,6 +3371,287 @@ static void arrayCopyDefault(TR::Node* node, uint8_t elementSize, TR::Register*
       }
    }
 
+/** \brief
+ *    Generate a store instruction
+ *
+ *  \param node
+ *     The tree node
+ *
+ *  \param addressReg
+ *     The base register for the destination mem address
+ *
+ *  \param index
+ *     The index for the destination mem address
+ *
+ *  \param valueReg
+ *     The value that needs be stored
+ *
+ *  \param size
+ *     The size of valueReg
+ *
+ *  \param cg
+ *     The code generator
+ *
+ */
+static void generateStoreInstructions(TR::Node* node, TR::Register* addressReg, int32_t index, TR::Register* valueReg, uint8_t size,  TR::CodeGenerator* cg)
```

Comment:
Sorry for the confusing, by my previous comment, I meant that probably we don't want to rename it to be something specific to arraycopy because this function is used in arraycopy and arrayset; this is for the second point in @0xdaryl 's comment.
As for the first point (plural problem), I will change it to singular.